scripts/check_compliance.py

```diff
@@ -588,6 +588,122 @@ def run(self):
             self.add_failure(msg.decode("utf-8"))
 
 
+class PyLint(ComplianceTest):
+    """
+    Runs pylint on all .py files, with a limited set of checks enabled
+    """
+    _name = "pylint"
+    _doc = "https://www.pylint.org/"
+
+    def run(self):
+        self.prepare(ZEPHYR_BASE)
+
+        # The commented-out checks below might be useful to enable, but
+        # currently trigger false positives
+        cmd = ("pylint --jobs=0 --score=n --persistent=n --disable=all "
+               "--enable "
+               "C0303,"  # Trailing whitspace
+               "C0304,"  # Final newline missing
+               "C0305,"  # Trailing newlines in file
+               "C0325,"  # Unnecessary parens after if/while, etc.
+               "E0100,"  # __init__ method is a generator
+               "E0101,"  # Explicit return in __init__
+               "E0102,"  # Function/method/class defined twice
+               "E0107,"  # Use of non-existent operators
+               "E0202,"  # Attribute named the same as method
+               "E0203,"  # Access to member before definition
+               "E0211,"  # Method has no argument
+               "E0213,"  # Method should have self as first argument
+               "E0221,"  # Interface resolved to x is not a class
+               "E0222,"  # Missing method from interface
+               #"E0601,"  # Variable referenced before assignment
```

Comment:
There's a comment at the top that explains why they're commented out, just in case you missed it.

I could move the list of disabled checks somewhere else though if you'd like.

Reply:
Also, I'm not 100% on having this be a whitelist instead of a blacklist (that one might become shorter too, though I haven't checked). With a blacklist, you might have to tweak it depending on `pylint` version though.

It's easy to tweak later at least.